src/scripts/main.js

```diff
@@ -1 +1,83 @@
-'use strict';
+/* eslint-disable prefer-promise-reject-errors */
+/* eslint-disable no-shadow */
+document.addEventListener('DOMContentLoaded', function () {
+  let leftClickOccurred = false;
+  let rightClickOccurred = false;
+
+  const firstPromise = new Promise((resolve, reject) => {
+    const leftClickHandler = () => {
+      resolve('First promise was resolved on a left click in the document');
+      document.removeEventListener('click', leftClickHandler);
+      clearTimeout(timeoutId);
+    };
+
+    document.addEventListener('click', leftClickHandler);
+
+    const timeoutId = setTimeout(() => {
+      reject('First promise was rejected in 3 seconds if not clicked');
+      document.removeEventListener('click', leftClickHandler);
+    }, 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    const clickHandler = () => {
+      resolve('Second promise was resolved');
+      document.removeEventListener('click', clickHandler);
+    };
+
+    document.addEventListener('click', clickHandler);
+    document.addEventListener('contextmenu', clickHandler);
+  });
+
+  const thirdPromise = new Promise((resolve) => {
+    const leftClickHandler = () => {
+      leftClickOccurred = true;
+      checkBothClicks();
+    };
+
+    const rightClickHandler = (event) => {
+      event.preventDefault();
```

Comment:
The `event.preventDefault()` is used here to prevent the default context menu from appearing on right-click. Ensure this behavior is desired, as it might interfere with user expectations for right-click actions.